src/objects/scales.ts

```diff
@@ -1,27 +1,23 @@
-import { ScaleLinear, scaleLinear } from 'd3'
-import { application } from '@/objects/application'
+import { ScaleLinear, scaleLinear, scaleTime, ScaleTime } from 'd3'
 import { emitter } from '@/objects/events'
 
-export let scaleX: ScaleLinear<number, number>
+export let scaleX: ScaleTime<number, number>
+export type ScaleX = typeof scaleX
+
 export let scaleY: ScaleLinear<number, number>
+export type ScaleY = typeof scaleY
 
-// these are the same for now but  ScaleXDomain will eventually be [Date, Date]
 type ScaleRange = [start: number, end: number]
-type ScaleXDomain = [start: number, end: number]
+type ScaleXDomain = [start: Date, end: Date]
 type ScaleYDomain = [start: number, end: number]
 
-export function createScales(): void {
-  createScaleX()
-  createScaleY()
-}
-
-export function createScaleX(): void {
-  const range: ScaleRange = [0, application.view.width]
-  const domain: ScaleXDomain = [0, 100]
 
-  scaleX = scaleLinear()
+export function startScales(): void {
+  scaleX = scaleTime()
+  scaleY = scaleLinear()
```

Comment:
scales are originally create without a domain or range. Because the stage doesn't exist yet. Then once the stage is created the initial ranges are set. 

A similar event will be used to set the domain based on the graph data.